components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaAuthMgtDAOImpl.java

```diff
@@ -0,0 +1,534 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.ciba.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.CibaCoreException;
+import org.wso2.carbon.identity.oauth.ciba.exceptions.ErrorCodes;
+import org.wso2.carbon.identity.oauth.ciba.model.CibaAuthCodeDO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Implementation of abstract DAO layer.
+ */
+public class CibaAuthMgtDAOImpl implements CibaAuthMgtDAO {
+
+    private static final Log log = LogFactory.getLog(CibaAuthMgtDAOImpl.class);
+
+    private CibaAuthMgtDAOImpl() {
+
+    }
+
+    private static CibaAuthMgtDAOImpl cibaAuthMgtDAOImplImplInstance = new CibaAuthMgtDAOImpl();
```

Comment:
Here we have already instantiated the DAO, then why do we need the singleton getInstance() method,

Reply:
Messed up writing **Double-Checked Locking with Singleton**. Will refactor.